pkg/networking/reconciler/reconciler.go

```diff
@@ -0,0 +1,38 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/controller"
+)
+
+type CiliumReconciler interface {
```

Comment:
Should this be called the `CiliumReconciler`? Let's say we have a new network addon that we change to, then we need to change this name here. Instead, shouldn't this be called a `CniReconciler` and the interface one level up the `NetworkReconciler`? That may not be the best naming still, but it seems like we don't have to call it cilium here

Reply:
I am also wondering if this shows whether we need these two networking reconcilers and only need one, where we call the cilium reconciler in the snow reconciler, but that's probably not the best because we want to separate responsibilities here. 

I think my concern is more the name of the interface is our reconcile signature may not look different anyways, and then we decide to split it up based on the names after we realize that the signature is going to be different.

Reply:
This interface is representing the Cilium reconciler, not the cni one. They just happen to have the same signature.

This first level cni reconciler is in charge of knowing which implementation of a reconciler needs to be called. We only support Cilium right now, but once we add docker support we will need to add kindnetd as well.

Reply:
Got it, so we will have `CiliumReconciler` and a `KindnetdReconciler` interface later which may have the same signature. I can't think of a reason that they would have a different signature which was why I was wondering if we wanted to generalize the interface more

Reply:
Yeah, they might have the same signature, that's fair. However, there is no benefit in enforcing that right now. And if we do and we are wrong, we would need to refactor a few things.

IMO it's probably safer to wait until we are sure we have an abstraction before actually trying to enforce one.